contents/naturalLanguage/nl.md

```diff
@@ -0,0 +1,169 @@
+Incidentally, this is how Microsoft Word checks for grammatical errors<sup>[6](#footnote6)</sup>.
+Any tree that cannot be parsed may be grammatically incorrect or difficult to understand.
+
+### Semantics
+With the knowledge of a sentence and its structure, the next step is to understand the meaning it conveys.
+To this end, the representation of the meaning of a sentence should be able to link language to concepts<sup>[7](#footnote7)</sup>.
+
+Take for example the sentence: 
+
+   > I wrote an article!
+
+In order to understand the meaning of the sentence, a few questions need to be answered:
+
+* Who is 'I'?
+* What is 'an article'? 
+* How are the two subjects related?
+
+An approach is to represent the meaning of a sentence as a relationship triple consisting of a *Subject*, *Object* and *Relation*.
+
+![Relation tuple](openIE.png "Constituency Parse") <br/>
+From this representation of the sentence, it is possible for a system to answer novel questions like "Who has written articles?" or "What the articles that 'I' have written".
+This technique, Open Information Extraction, is utilized by IBM's Watson to defeat human players in the American game show Jeopardy!<sup>[8](#footnote8)</sup>
+
+### Relations 
+Relations are crucial to the understanding of natural languages.
+Consider the example:
+
+   > Geoffrey bought Mary a ring. They have been dating for months.
+
+What is the event that prompted Geoffrey to buy a ring? 
+The word _proposal_, while not stated in the example, implicitly links the two events together.
+While humans are able to draw the connections instinctually, it is not easy to formalize this process computationally.
+
+There are attempts to create semantic _ontologies_ for natural language, groupings of words by their meanings.
+**WordNet** is one such attempt, a handcrafted database that classifies words by concepts that they express<sup>[9](#footnote9)</sup>.
+
+![WordNet](wordNet.png "WordNet") <br/>
+
+More recently, techniques like **word2vec**<sup>[10](#footnote10)</sup> are able to model words and phrases as vectors of real numbers. 
+This ability to quantify and categorize semantic similarity allows computers to infer beyond just synonyms.
+
+![word2vec](word2vec.png "Word2Vec") <br/>
+
+A simplified visualization of words in vector space shows that not only are words related in meaning close together, their antonyms are clustered together as well<sup>[11](#footnote11)</sup>. 
+This implies that word vectors can be somewhat meaningfully combined by just using simple vector addition.
+
+Word2vec has been used to analyze behaviour on e-commerce sites as human do not tend to browse for random items but purchase items that are related.
+
+### Applications of NLP
+
+#### Input methods editors, e.g. Chinese, Japanese and Korean
+Modern input method editors(IMEs) do more than simply translating input to output in the respective language.
```

Comment:
Might be good to elaborate on what Method Input Editors are, since it's introduced here for the first time